extract_unique_sensors_01.py

```diff
@@ -13,45 +17,55 @@ def create_parser():
 
     parser = argparse.ArgumentParser(description='Extract unique sensors')
     parser.add_argument('--file_name', help='File to extract unique sensors from.')
-    parser.add_argument('--url', help='A reference to SensorList.txt that specifies its ' 
-                        'location on a computer network.') 
+    parser.add_argument('--url', help='A reference to SensorList.txt that specifies its'
+                        'location on a computer network.')
     parser.add_argument('--kat_sensor', required=True, help='Name of unique sensor')
+    parser.add_argument('-v', metavar='verbosity', type=int, default=2, help='Logging'
+                        'verbosity: 0 -critical, 1- error, 2 -warning, 3 -info, 4 -debug')
+
     args = parser.parse_args()
     return args
 
+
 def read_sensors(file_name, url):
     """This function opens, reads and extracts the contents of the file_name.If the file_name
-    does not exist,with try will attempt to read the file_name from the url expecting it to be
-    there. If not, catch exception and perform a fallback.
+    does not exist,with try will attempt to read the file_name from the url expecting it
+    to be there. If not, catch exception and perform a fallback.
 
     Params
     ------
-    file_name: str        
+    file_name: str
         The actual name of the file.
-    url: str                       
+    url: url
         The location of SensorList.txt on the internet.
-                                                    
+
     Returns
     -------
     sensors_data: list
-        list of sensors. 
-    """                 
+        list of sensors.
+    """
     assert isinstance(file_name, str), "Expected a string."
-    assert isinstance(url, str), "Expected a string."                                          
+    assert isinstance(url, str), "Expected a string."
     sensors_data = []
     if file_name and Path(file_name).exists():
-        file_path = Path(file_name)           
+        file_path = Path(file_name)
         with open(file_path) as lines:
             sensors_data = lines.readlines()
-    elif url:                               
-        try: 
+    elif url:
+        try:
             sensors_url = urlopen(url)
             sensors_data = sensors_url.readlines()
-        except Exception:                         
-            print("Failed to retrieve sensor data")
-    return sensors_data  
- 
-def extract_all_sensors(sensors_data, sensor_name): 
+            logging.debug("Using the url.")
+            logging.debug("Using the filename")
+
+        except Exception:
+            logging.exception(
+                "Failed to retrieve sensor data. Unkown url type or filename.")
```

Comment:
Your closing bracket should be inline with the start of your code, not sure if that makes sense.
I also suggested some typos fixes.

```suggestion
            logging.exception(
                "Failed to retrieve sensor data. Unknown URL type or filename."
            )
````